File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    let bestBuy = prices[0];
    let maxprofit = 0;
    for(let i=0; i<prices.length; i++){
        if(prices[i] > bestBuy){
            maxprofit = Math.max(maxprofit, prices[i] - bestBuy)
        }
        bestBuy = Math.min(bestBuy, prices[i]);
    }
    return maxprofit;
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        bestbuy = prices[0]
        maxprofit = 0
        i = 0

        while(i < len(prices)):
            bestbuy = min(bestbuy, prices[i])
            if(prices[i] > bestbuy):
                maxprofit = max(maxprofit, prices[i] - bestbuy)

            i+=1
        return maxprofit
                